### ISSUE TITLE
REFL005 should not warn when BindingFlags.NonPublic is used and internals aren't known

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#120]
0.1.9-dev

Here's an example where there is an internal overload being invoked and REFL005 can only see the public overload:

```c#
typeof(Control).GetMethod(nameof(Control.CreateControl), BindingFlags.Instance | BindingFlags.NonPublic, null, new[] { typeof(bool) }, null)
```

> REFL005 There is no member matching the filter. Expected: BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly.
